html/user-activation/resources/utils.js

```diff
@@ -16,20 +16,26 @@ function getEvent(eventType) {
   });
 }
 
-
-// Returns a Promise which is resolved with a "true" iff transient activation
-// was available and successfully consumed.
-//
-// This function relies on Fullscreen API to check/consume user activation
-// state.
-async function consumeTransientActivation() {
-  try {
-    await document.body.requestFullscreen();
-    await document.exitFullscreen();
-    return true;
-  } catch(e) {
-    return false;
+/**
+ *
+ * @param {Window} context
+ * @returns {Promise<Boolean>} resolved with a true if transient activation is consumed.
+ */
+async function consumeTransientActivation(context = window) {
+  if (!context.navigator.userActivation.isActive) {
+    throw new Error(
+      "User activation is not active so can't be consumed. Something is probably wrong with the test."
+    );
+  }
+  if (test_driver?.consume_user_activation) {
+    return test_driver.consume_user_activation(context);
+  }
```

Comment:
This seems to regress tests in every browser, e.g. https://wpt.fyi/results/html/user-activation?diff&filter=ADC&run_id=6320849824514048&run_id=5200899105882112

Even if wptrunner didn't cause these to ERROR, and this got the `throw new Error("consume_user_activation() is not implemented by testdriver-vendor.js")`, this would still regress tests.

Reply:
Ok, I'll be more kind there and use the fallback.